packages/babel-traverse/test/conversion.js

```diff
@@ -51,6 +51,14 @@ describe("conversion", function () {
       expect(generateCode(rootPath)).toBe("() => {\n  return false;\n};");
     });
 
+    it("preserves arrow function body's context", function () {
```

Comment:
Do you need another test, or is this enough? (Since there is the to-do item in the PR description)

Reply:
This test is copied from the `replaceWith` test above, see also https://github.com/babel/babel/pull/12391#discussion_r529205110. The second commit fixes another regression introduced in #12302 that is not yet reported.

Reply:
> I can work on it if needed.

Appreciated. Feel free to push 😄. I am still trying to figure out how we should test the `pathCache` behaviour in `@babel/traverse`.